pkg/prove/proof.go

```diff
@@ -203,3 +206,105 @@ func splitIntoRows(squareSize uint64, s []shares.Share) [][]shares.Share {
 	}
 	return rows
 }
+
+// SharesInclusion generates an nmt inclusion proof for a set of shares to the data root.
+// expects the shares range to be pre-validated.
+// Note: only supports inclusion proofs for shares belonging to the same namespace.
+func SharesInclusion(
+	codec rsmt2d.Codec,
+	allRawShares []shares.Share,
+	squareSize uint64,
+	namespaceID namespace.ID,
+	startShare uint64,
+	endShare uint64,
+) (types.SharesProof, error) {
+	startRow := startShare / squareSize
+	endRow := endShare / squareSize
+	startLeaf := startShare % squareSize
+	endLeaf := endShare % squareSize
+
+	eds, err := rsmt2d.ComputeExtendedDataSquare(
+		shares.ToBytes(allRawShares),
+		appconsts.DefaultCodec(),
+		wrapper.NewConstructor(squareSize),
+	)
+	if err != nil {
+		return types.SharesProof{}, err
+	}
+
+	// create the binary merkle inclusion proof, for all the square rows, to the data root
```

Comment:
[Note For reviewers]
Do you think I should put the part generating the eds and the binary merkle proof in a separate function? 
Same for the part generating the NMT? (as it is almost a duplicate of the tx inclusion generation).
I am keeping everything in the same function to reduce jumps and for better clarity.

Reply:
like this? [d0110a1](https://github.com/celestiaorg/celestia-app/pull/1042/commits/d0110a177532ee1f7c2ae08e8e08e70f81247094)